client/my-sites/stats/hooks/use-stats-purchases.tsx

```diff
@@ -121,25 +124,26 @@ export default function useStatsPurchases( siteId: number | null ) {
 	const hasLoadedSitePurchases = useSelector( hasLoadedSitePurchasesFromServer );
 
 	// Determine whether a product is owned.
-	// TODO we need to do plan check as well, because Stats products would be built into other plans.
-	const isFreeOwned = useMemo( () => {
-		return isProductOwned( sitePurchases, PRODUCT_JETPACK_STATS_FREE );
-	}, [ sitePurchases ] );
+	const isFreeOwned = useSelector( ( state ) => {
+		return siteHasFeature( state, siteId, FEATURE_STATS_FREE );
```

Comment:
Having the feature FEATURE_STATS_FREE doesn't necessarily mean the site has the product, does it?

Reply:
The feature `FEATURE_STATS_FREE` means benefits for legacy WPCOM sites, which might not correspond to our previous usage. Context: D167647-code.